cdisc_rules_engine/services/data_services/base_data_service.py

```diff
@@ -132,15 +132,31 @@ def concat_split_datasets(
             func_to_call, dataset_names, **kwargs
         )
 
-        # concat datasets
-        full_dataset: pd.DataFrame = pd.concat(
-            [dataset for dataset in datasets],
-            ignore_index=True,
-        )
+        full_dataset = pd.DataFrame()
+        for dataset in datasets:
+            if "RDOMAIN" in dataset.columns:
+                supp_merged_dataset = self.merge_supp_dataset(full_dataset, dataset)
+                full_dataset = supp_merged_dataset
```

Comment:
is supp_merged_dataset variable necessary?  Could we just make full_datase equal to the function call on line 138?